keylime/tpm/tpm_main.py

```diff
@@ -1576,6 +1583,32 @@ def __add_boot_aggregate(log: dict) -> None:
                 except Exception:
                     pass
 
+    @staticmethod
+    def __unescape_eventlog(log: dict):
+        """
+        Newer versions of tpm2-tools escapes the YAML output and includes the null byte.
+        This converts it back to an unescaped string.
+        Example:
+            '"MokList\\0"' -> 'MokList'
+        """
+        if tpm.tools_version in ["3.2", "4.0", "4.2"]:
+            return
+
+        def recursive_unescape(data):
+            if isinstance(data, str):
+                if data.startswith('"') and data.endswith('\\0"'):
```

Comment:
I am wondering 2 stuff:

1. maybe the datatype is wrong, and should be byte instead of str
2. is there ant guarantee that the null byte will appear only in the last position? I am thinking on other assets that are measured and part of the event log that will contain the null byte in the middle.  In this case the parser should be more genering.

Reply:
1. It is actually a str, because it is part of the YAML output of tpm2-tools which is converted to a dict.
2. Kinda, there should be always a final trailing one. If not someone writes malformed strings. See this commit for more info: https://github.com/tpm2-software/tpm2-tools/commit/c78d258b2588aee535fd17594ad2f5e808056373

Reply:
Thanks for the pointer!

```
- All NUL bytes, including the final trailing NUL
    are displayed in escaped format.
```

Glazing the code seems to be more escapes ("\a", "\n", "\t", ...) at any point in the byte array.

IMHO would make sense to reflect this into the parser too.  Something trivial, like a search-replace.  The cost will be `10 * n` (10 escape character for a string of length n), as we will repeat the search-replace 10 times. So is still `O(n)`, so kind of OK.

Reply:
Yeah we should do that. My idea is the following

1. Check if string is in quotes
2. Remove all the escaped characters using search-replace. (I still need check if simple search-replace is enough)
3. Remove trailing null if present

Reply:
I now implemented the idea.